pinot-core/src/main/java/org/apache/pinot/core/data/table/IndexedTable.java

```diff
@@ -37,6 +42,8 @@
  */
 @SuppressWarnings({"rawtypes", "unchecked"})
 public abstract class IndexedTable extends BaseTable {
```

Comment:
I think it'd be good to other pool constructor and:
- grant pool threads reasonable names to make identifying them in thread stack dumps easier
- make them deamon threads so they don't affect jvm shutdown (especially that pool is not closed anywhere)[](url)